transformer/dockerfilegenerator/dotnetcoredockerfilegenerator.go

```diff
@@ -0,0 +1,492 @@
+func getPublishProfile(publishProfileFilesPath []string, serviceName, baseDir string) string {
+	var publishProfileFilesRelPath []string
+	for _, publishProfileFilePath := range publishProfileFilesPath {
+		if publishProfileFileRelPath, err := filepath.Rel(baseDir, publishProfileFilePath); err == nil {
+			publishProfileFilesRelPath = append(publishProfileFilesRelPath, publishProfileFileRelPath)
+		}
+	}
+	if len(publishProfileFilesRelPath) == 1 {
+		return publishProfileFilesRelPath[0]
+	} else if len(publishProfileFilesRelPath) > 1 {
+		return qaengine.FetchSelectAnswer(common.ConfigServicesKey+common.Delim+serviceName+common.Delim+common.ConfigPublishProfileForServiceKeySegment, fmt.Sprintf("Select the publish profile to be used for the service %s :", serviceName), []string{fmt.Sprintf("Selected publish profile will be used for publishing the service %s", serviceName)}, publishProfileFilesRelPath[0], publishProfileFilesRelPath)
+	}
+	return ""
+}
+
+// findPublishProfiles returns the publish profiles of the cs project
+func findPublishProfiles(csprojFilePath string) []string {
+	publishProfileFiles, err := common.GetFilesByExt(filepath.Dir(csprojFilePath), []string{".pubxml"})
+	if err != nil {
+		logrus.Debugf("Error while finding publish profile (.pubxml) files %s", err)
+		return []string{}
+	}
+	return publishProfileFiles
+}
+
+// parsePublishProfileFile parses the publish profile to get the PublishUrl
+func parsePublishProfileFile(publishProfileFilePath string) (string, error) {
+	var publishUrl string
+	publishProfile := PublishProfile{}
+	xmlFile, err := os.Open(publishProfileFilePath)
+	if err != nil {
+		logrus.Errorf("Could not open the publish profile file: %s", err)
+		return "", err
+	}
+	defer xmlFile.Close()
+	byteValue, err := io.ReadAll(xmlFile)
+	if err != nil {
+		logrus.Errorf("Could not read the publish profile file: %s", err)
+		return "", err
+	}
+	err = xml.Unmarshal(byteValue, &publishProfile)
+	if err != nil {
+		logrus.Errorf("Could not parse the publish profile file %s", err)
+		return "", err
+	}
+	separator := fmt.Sprintf("%c", os.PathSeparator)
+	if publishProfile.PropertyGroup.PublishUrl != "" {
+		publishUrl = strings.ReplaceAll(publishProfile.PropertyGroup.PublishUrl, `\`, separator)
+	} else if publishProfile.PropertyGroup.PublishUrlS != "" {
+		publishUrl = strings.ReplaceAll(publishProfile.PropertyGroup.PublishUrlS, `\`, separator)
+	}
+	return publishUrl, nil
+}
+
+// findDefaultCsprojForQA returns the index of the first csproj which has publish profile
+func findDefaultCsprojForQA(csprojFilesPath []string) (int, []string) {
+	for i, csprojFilePath := range csprojFilesPath {
+		publishProfileFiles, err := common.GetFilesByExt(filepath.Dir(csprojFilePath), []string{".pubxml"})
+		if err != nil {
+			logrus.Debugf("Error while finding publish profile (.pubxml) files %s", err)
+		}
+		if len(publishProfileFiles) != 0 {
+			return i, publishProfileFiles
+		}
+	}
+	return 0, []string{}
+}
```

Comment:
Done. I am returning the index of the csproj file to avoid computing the relative path of the same csproj file twice in the function getCsprojFileForService.